custom/abdm/milestone_one/utils/response_util.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework.response import Response
+
+from rest_framework.status import HTTP_200_OK, HTTP_500_INTERNAL_SERVER_ERROR, HTTP_400_BAD_REQUEST
+
+success_response_keys = ["txnId", "healthIdNumber", "auth_methods", "token"]
+
+
+def get_response(response_data):
+    if not response_data:
+        return Response({"error": "No valid response found"}, status=HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
We have seen in the past that response status set by ABDM is not very assuring but they rather rely on sending HTTP status code in a dict. This is the reason we are setting code in our object by extracting it from their response json. This is done in `_parse_response` of same class.

> Also, no response could simply be consumed within the check for success response keys.

Didn't get this part completely.

Reply:
> We could simplify this a bit by returning response from get_response_http_post and not just the json.

I should have said response "object" here. What I meant was that we are passing the response json here, but could pass the response object and use that to get the code instead of populating it separately.

> Didn't get this part completely.

There is a mistype there from me. I can't remember that as of now, but seems okay to ignore it for now.

Reply:
> What I meant was that we are passing the response json here, but could pass the response object and use that to get the code instead of populating it separately.
This is what is problematic from ABDM side. Their response object's status code and status code in response json are not always in sync and in such cases, code mentioned in the JSON was correct one(I am not able to recall the exact scenario). Due to this reason we are neglecting the response object.

Reply:
> There is a mistype there from me. I can't remember that as of now, but seems okay to ignore it for now.

ok